sub/sd_ass.c

```diff
@@ -359,10 +359,14 @@ static void decode(struct sd *sd, struct demux_packet *packet)
             filter_and_add(sd, &pkt2);
         }
         if (ctx->duration_unknown) {
-            for (int n = 0; n < track->n_events - 1; n++) {
-                if (track->events[n].Duration == UNKNOWN_DURATION * 1000) {
-                    track->events[n].Duration = track->events[n + 1].Start -
-                                                track->events[n].Start;
+            for (int n = 2; n <= track->n_events; n++) {
```

Comment:
Why not just count down here with this for loop? `for (n = track->n_events; n >= 2; n--) {`

Reply:
Will conform right away.